plan/physical.go

```diff
@@ -123,6 +123,54 @@ func (pp *physicalPlanner) Plan(ctx context.Context, spec *Spec) (*Spec, error)
 	return transformedSpec, nil
 }
 
+func ValidateAttributes(plan *Spec) error {
+	err := plan.BottomUpWalk(func(pn Node) error {
+		ppn, ok := pn.(*PhysicalPlanNode)
+		if !ok {
+			return fmt.Errorf("invalid physical query plan; found logical operation \"%v\"", pn.ID())
+		}
+
+		for key, attr := range ppn.RequiredAttrs {
+			for _, pred := range ppn.Predecessors() {
+				ppred := pred.(*PhysicalPlanNode)
+				predAttr, ok := ppred.OutputAttrs[key]
+				if !ok {
+					return fmt.Errorf("invalid physical query plan; attribute \"%v\" required by "+
+						"\"%v\" is missing from predecessor \"%v\"", key, ppn.id, ppred.id)
+				}
+				if attr != predAttr {
```

Comment:
If we are going to use a simple equality check here, that has some implications.  There could also be a `Equals` method or something. Seems fine for now, we might want to revisit it later.

Reply:
could I trouble you to change this error and the other errors in this file to use the modern format for generating errors?
```go
return flux.Error{
  code: codes.Internal, // flux/codes
  Msg: "...",
}
```
This message would indicate a bug in the planner or in planner rules, so it would be good to explicitly make it internal.

Reply:
Done.

Reply:
I upgraded all errors in `plan/physical.go` to use `flux.Error`.